pkg/cmd/cli/install/install.go

```diff
@@ -246,9 +253,30 @@ This is useful as a starting point for more customized installations.
 
 // Run executes a command in the context of the provided arguments.
 func (o *InstallOptions) Run(c *cobra.Command, f client.Factory) error {
+	// Find the kube-apiserver group apiextensions.k8s.io preferred API version
+	clientset, err := f.KubeClient()
+	if err == nil {
+		// kubeconfig available
+		discoveryHelper, err := velerodiscovery.NewHelper(clientset.Discovery(), &logrus.Logger{})
+		if err == nil {
+			// kubernetes apiserver available
+			gvr, _, err := discoveryHelper.ResourceFor(
+				schema.GroupVersionResource{
+					Group:    "apiextensions.k8s.io",
+					Resource: "customresourcedefinitions",
+				})
+			if err != nil {
+				return err
+			}
+
+			// Update the group apiextensions.k8s.io preferred API version
+			o.CRDsVersion = gvr.Version
```

Comment:
With this, the API server preferred version will always take precedence over the setting provided by the user with the `--crds-version` flag. Is there a case where the user may want to explicitly use v1beta1 when the preferred version is v1?

Also, in the case where Velero can't find the server preferred version, we should output a warning or log to state that the preferred version couldn't be found and that instead we will use `v1` as the default and that it can be changed with the `--crds-version` flag. Otherwise, I think we may encounter users who think that the CRD installation is broken on pre-1.16 clusters if the discovery fails.